### PR TITLE
Refactor 

### PR DESCRIPTION
**Pull request by heron2014**
## Ready for review

Related to #524 & #17 & #176  

@nelsonic please have a look at this PR, thank you

### CODE REVIEW

**Comment by nelsonic** [comment]
@heron2014 looking good so far. once you've merged master in & finished off, feel free to assign & ping me. 👍

**Comment by heron2014** [comment]
@nelsonic I think this is good to go.

**Comment by heron2014** [comment]
@SimonLab if you can also run this branch and check if all works as expected as I have changes the few handlers. Thank you

File: assets/js/client_dashboard.js [hunk 1]
```diff
@@ -1,105 +1,120 @@
...
+              currentStageDOM.removeChild(currentCandidateDOM);
+              siblingCurrentStage.appendChild(currentCandidateDOM);
+            }
+
+          }
 
-        if (JSON.parse(response).code === 500) {
-          currentCandidateDOM.querySelector('.error-msg').style.display = 'block';
+          if (JSON.parse(response).code === 500) {
+            showErrorMessage(currentCandidateDOM);
```
Comment (nelsonic) [inline assets/js/client_dashboard.js]: 👍

**Comment by nelsonic** [comment]
Nice work @heron2014 👍

**Pull request merged**